unit/api/reward_resource.py

```diff
@@ -0,0 +1,39 @@
+from typing import Union, List
+
+from unit.api.base_resource import BaseResource
+from unit.models import UnitResponse, UnitError
+from unit.models.reward import RewardDTO, ListRewardsParams, CreateRewardRequest
+
+from unit.models.codecs import DtoDecoder
+
+
+class RewardResource(BaseResource):
```

Comment:
Has not been tested yet, will likely uncover some bugs in the `Reward` related classes later this week